ultralight-java-databind/src/main/java/net/labymedia/ultralight/utils/JavascriptUtils.java

```diff
@@ -0,0 +1,106 @@
+package net.labymedia.ultralight.utils;
+
+import net.labymedia.ultralight.UltralightDatabind;
+import net.labymedia.ultralight.UltralightDatabindConfiguration;
+import net.labymedia.ultralight.containers.GenericValueContainer;
+import net.labymedia.ultralight.javascript.JavascriptClass;
+import net.labymedia.ultralight.javascript.JavascriptContext;
+import net.labymedia.ultralight.javascript.JavascriptType;
+import net.labymedia.ultralight.javascript.JavascriptValue;
+
+import java.text.DateFormat;
+import java.text.ParseException;
+import java.text.SimpleDateFormat;
+import java.util.Date;
+
+public final class JavascriptUtils {
+    private static DateFormat JAVASCRIPT_DATE_FORMAT = new SimpleDateFormat("EEE MMM d yyyy HH:mm:ss 'GMT'Z '('z')'");
+
+    private JavascriptUtils() { }
+
+    public static JavascriptValue toJavascriptValue(final JavascriptContext context, final Object javaObject) {
+        return toJavascriptValue(context, javaObject.getClass(), javaObject);
+    }
+
+    public static JavascriptValue toJavascriptValue(final JavascriptContext context, final Class<?> javaClass, final Object javaObject) {
+        if (javaObject == JavascriptType.UNDEFINED) return context.makeUndefined();
+        if (javaObject == null) return context.makeNull();
+
+        if (javaClass == Short.class
+                || javaClass == Integer.class
+                || javaClass == Long.class
+                || javaClass == Float.class
+                || javaClass == Double.class
+                || javaClass == Byte.class
+                || javaClass == short.class
+                || javaClass == int.class
+                || javaClass == long.class
+                || javaClass == float.class
+                || javaClass == double.class
+                || javaClass == byte.class) return context.makeNumber((double) javaObject);
+        if (javaClass == Character.class) return context.makeString(String.valueOf((char) javaObject));
+        if (javaClass == Boolean.class) return context.makeBoolean((boolean) javaObject);
+        if (javaClass == String.class) return context.makeString((String) javaObject);
+        if (javaClass == Date.class) {
```

Comment:
Not sure if the checks are correct here. For example, the ``if(javaClass == Character.class)`` then casts the value to a ``char``. However, a ``java.lang.Character`` can't be casted to a char, but the value can be retrieved using ``java.lang.Character#charValue()``. 

Same goes for the classes above, but a ``Number.class.isAssignableFrom(javaClass)`` should be enough, the value can then be retrieved as ``java.lang.Number#doubleValue()``.